server/src/main/java/org/opensearch/action/termvectors/TransportTermVectorsAction.java

```diff
@@ -87,15 +90,30 @@ public TransportTermVectorsAction(
 
     @Override
     protected ShardIterator shards(ClusterState state, InternalRequest request) {
+
+        final String preference;
+        // route realtime TermVector requests when segment replication is enabled to primary shards,
+        // iff there are no other preferences/routings enabled for routing to a specific shard
```

Comment:
We are not considering routings in shouldForcePrimaryRouting. Update comment to - 
```suggestion
        // iff there are no other preferences enabled for routing to a specific shard
```